python/ray/serve/common.py

```diff
@@ -39,3 +39,34 @@ def __init__(self,
         self.deployer_job_id = deployer_job_id
         # The time when this deployment was deleted.
         self.end_time_ms = end_time_ms
+
+
+@dataclass
+class ReplicaName:
+    # Contextual input sources
+    deployment_tag: BackendTag
+    replica_suffix: str
+    # Generated replica name with fixed naming rule
```

Comment:
What is the fixed naming rule? needs more context.

Reply:
delimiter and format should be stable, consistent and not conflicting with others (like we put ":" in actor naming when combining with controller). I need another pass at this class for readability ...